pontoon/base/tests/test_models.py

```diff
@@ -0,0 +1,75 @@
+from django_nose.tools import assert_equal
+
+from pontoon.base.models import Translation
+from pontoon.base.tests import (
+    assert_redirects,
+    EntityFactory,
+    LocaleFactory,
+    ProjectFactory,
+    ResourceFactory,
+    TranslationFactory,
+    UserFactory,
+    TestCase
+)
+from pontoon.base.utils import aware_datetime
+
+
+class LocaleTests(TestCase):
+    def test_latest_activity_translated(self):
+        """
+        If latest activity in Translation QuerySet is translation submission,
+        return submission date and user.
+        """
+        user0 = UserFactory.create()
+        user1 = UserFactory.create()
+
+        translation0 = TranslationFactory.create(
+            date=aware_datetime(1970, 1, 1),
+            user=user0,
+            approved_date=aware_datetime(1970, 1, 1),
+            approved_user=user0
+        )
+        translation1 = TranslationFactory.create(
+            date=aware_datetime(1970, 1, 2),
+            user=user1,
+            approved_date=aware_datetime(1970, 1, 2),
+            approved_user=user1
+        )
+
+        translations = Translation.objects.filter(id__in=[translation0.id, translation1.id])
```

Comment:
No need to filter here. The tables are cleared between tests, so the only translations that exist are the ones you create. `Translations.objects.all().latest_activity()` is equivalent!